gr-utils/python/blocktool/tests/test_blocktool.py

```diff
@@ -0,0 +1,187 @@
+#
+# Copyright 2019 Free Software Foundation, Inc.
+#
+# This file is part of GNU Radio
+#
+# GNU Radio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3, or (at your option)
+# any later version.
+#
+# GNU Radio is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with GNU Radio; see the file COPYING.  If not, write to
+# the Free Software Foundation, Inc., 51 Franklin Street,
+# Boston, MA 02110-1301, USA.
+#
+""" unittest for gr-blocktool api """
+
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import os
+import unittest
+import warnings
+try:
+    import pygccxml
+    SKIP_BLOCK_TEST = False
+except:
+    SKIP_BLOCK_TEST = True
+
+try:
+    import apt
+    CACHE = apt.cache.Cache()
+    CACHE.open()
+
+    PKG = CACHE['castxml']
+    if PKG.is_installed:
+        SKIP_BLOCK_TEST = False
+    else:
+        SKIP_BLOCK_TEST = True
+except:
+    SKIP_BLOCK_TEST = True
+
+from jsonschema import validate
+
+from blocktool.core.parseheader import BlockHeaderParser
+from blocktool.core.base import BlockToolException
+from blocktool.core import Constants
+from blocktool.core.outputschema import PARSED_OUTPUT_SCHEME
+
+
+class TestBlocktoolCore(unittest.TestCase):
+    """ The Tests for blocktool core """
+
+    def __init__(self, *args, **kwargs):
+        super(TestBlocktoolCore, self).__init__(*args, **kwargs)
+        self.current_folder = os.path.abspath(os.path.dirname(__file__))
+        self.test_dir = os.path.abspath(os.path.join(
+            self.current_folder, '../../../../gr-analog/include/gnuradio/analog'))
+        self.target_file = os.path.abspath(
+            os.path.join(self.test_dir, 'agc2_cc.h'))
+        self.impl_dir = os.path.abspath(os.path.join(self.current_folder,
+                                                     '../../../../gr-analog/lib'))
+        self.modname = os.path.basename(os.path.dirname(
+            os.path.dirname(os.path.dirname(os.path.dirname(self.target_file)))))
+        self.gr = self.modname.split('-')[0]
+        self.module = self.modname.split('-')[-1]
+
+    def is_int(self, number):
```

Comment:
this is a general UnitTest but seems unrelated to the specifics of BlockTool

Reply:
I didn't get it. All the __init__ variables defined above are required to check the Blocktool raised exceptions as in the validate function of `core/parseheader.py`